app/code/local/LoyaltyLion/Core/Model/Observer.php

```diff
@@ -4,23 +4,38 @@ class LoyaltyLion_Core_Model_Observer {
 
   private $client;
   private $session;
+  private $tokensByStoreId = [];
 
   public function __construct() {
-    $this->token = Mage::getStoreConfig('loyaltylion/configuration/loyaltylion_token');
-    $this->secret = Mage::getStoreConfig('loyaltylion/configuration/loyaltylion_secret');
-
     if (!$this->isEnabled()) return;
 
     $this->client = Mage::helper('loyaltylion/client')->client();
 
     $this->session = Mage::getSingleton('core/session');
   }
 
-  private function isEnabled() {
-    if (empty($this->token) || empty($this->secret)) return false;
+  private function isEnabled($storeId = null) {
+    if (is_null($storeId)) $storeId = Mage::app()->getStore()->getId();
```

Comment:
When in the UI this will return the store that the customer is currently browsing, this makes it fairly safe for these cases to pass null to this method (ideally we shouldn't, where we can we should pass an Id).

In an admin scenario this method returns 0 or the id of the default magento store created on install. In these cases we always want to pass and Id. We can fetch this from the order / customer instance passed to our controller actions

Reply:
We could also not do this as passing null is safe through to `getStoreConfig` as it's function definition defaults the third parameter to null.

Reply:
Your second line would be great to preserve as part of the comment!

I'm uneasy about this implementation in two ways:

A) being explicit is good, but how do we know that `Mage::app()->getStore()->getId()` replicates the current implicit behaviour in all contexts (and over time)? 

B) Accidentally passing `null`: in certain contexts, we should expect to call this function in relation to models that haven't actually been persisted to the db yet; for reasons I don't fully understand, for e.g. an order create event before the order has been persisted. As a result - while logically, `$order->getStoreId()` should probably always return a non-null value I'd consider it plausible that it could return null in some situations. It feels like the safer thing to do here would be to return `false` and have a separate method for other contexts (UI). What do you think?